tests/unit/lms/services/assignment_test.py

```diff
@@ -239,55 +239,52 @@ def test_is_member(self, svc, db_session):
 
-    def test_get_assignments(self, svc, db_session):
-        assert db_session.scalars(svc.get_assignments()).all()
-
-    def test_get_assignments_by_course(self, svc, db_session, assignment):
+    @pytest.mark.parametrize("instructor_h_userid", [True, False])
+    @pytest.mark.parametrize("course_id", [True, False])
+    @pytest.mark.parametrize("h_userids", [True, False])
+    def test_get_assignments(
+        self,
+        svc,
+        db_session,
+        instructor_h_userid,
+        assignment,
+        with_assignment_noise,
+        course_id,
+        h_userids,
+    ):
+        factories.User()
         course = factories.Course()
-        factories.AssignmentGrouping.create(assignment=assignment, grouping=course)
-        db_session.flush()
-
-        assert db_session.scalars(svc.get_assignments(course_id=course.id)).all() == [
-            assignment
-        ]
-
-    def test_get_assignments_with_instructor_h_userid(self, svc, db_session):
-        factories.User()  # User not in assignment
-        assignment = factories.Assignment()
         user = factories.User()
-        lti_role = factories.LTIRole(scope=RoleScope.COURSE)
+        lti_role = factories.LTIRole(scope=RoleScope.COURSE, type=RoleType.INSTRUCTOR)
         factories.AssignmentMembership.create(
             assignment=assignment, user=user, lti_role=lti_role
         )
-        # Other membership record, with a different role
         factories.AssignmentMembership.create(
             assignment=assignment, user=user, lti_role=factories.LTIRole()
         )
-
+        factories.AssignmentGrouping.create(assignment=assignment, grouping=course)
         db_session.flush()
 
-        assert db_session.scalars(svc.get_assignments(user.h_userid)).all() == [
-            assignment
-        ]
+        query_parameters = {}
 
-    def test_get_assignments_with_h_userids(self, svc, db_session):
-        factories.User()  # User not in assignment
-        assignment = factories.Assignment()
-        user = factories.User()
-        lti_role = factories.LTIRole(scope=RoleScope.COURSE)
-        factories.AssignmentMembership.create(
-            assignment=assignment, user=user, lti_role=lti_role
-        )
-        # Other membership record, with a different role
-        factories.AssignmentMembership.create(
-            assignment=assignment, user=user, lti_role=factories.LTIRole()
-        )
+        if instructor_h_userid:
+            query_parameters["instructor_h_userid"] = user.h_userid
 
-        db_session.flush()
+        if course_id:
+            query_parameters["course_id"] = course.id
 
-        assert db_session.scalars(
-            svc.get_assignments(h_userids=[user.h_userid])
-        ).all() == [assignment]
+        if h_userids:
+            query_parameters["h_userids"] = [user.h_userid]
+
+        query = svc.get_assignments(**query_parameters)
+
+        if not query_parameters:
```

Comment:
Improve the assert here, check it matches the test created in this test (`assignment`) + the other ones in the DB (`with_assignment_noise`).